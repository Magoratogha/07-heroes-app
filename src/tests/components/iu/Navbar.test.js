import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import { AuthContext } from "../../../auth/authContext";
import Navbar from "../../../components/ui/Navbar";
import { types } from "../../../types/types";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockNavigate
}));

describe('Test on <Navbar />', () => {

    const user = userEvent.setup();

    const contextValue = {
        user: {
            name: "Batman",
            isLogged: true
        },
        dispatch: jest.fn()
    }

    it('should render properly', () => {
        const { container } = render(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={['/']}>
                    <Navbar />
                </MemoryRouter>
            </AuthContext.Provider>
        );
        expect(container).toMatchSnapshot();
        expect(screen.getByText('Batman', { selector: 'span' })).toBeInTheDocument();
    });

    it('should call navigate and dispatch functions when user logout', async() => {
        render(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={['/']}>
                    <Navbar />
                </MemoryRouter>
            </AuthContext.Provider>
        );
        const logoutButton = screen.getByText('Logout', { selector: 'button' });
        await user.click(logoutButton);
        expect(mockNavigate).toHaveBeenCalledWith('/login', { replace: true });
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.logout
        });
    });
});