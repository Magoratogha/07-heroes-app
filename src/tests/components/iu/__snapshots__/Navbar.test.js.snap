// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Navbar /> should render properly 1`] = `
<div>
  <nav
    class="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/marvel"
      >
        HeroesApp
      </a>
      <div
        class="navbar-collapse"
      >
        <div
          class="navbar-nav"
        >
          <a
            class="nav-item nav-link"
            href="/marvel"
          >
            Marvel
          </a>
          <a
            class="nav-item nav-link"
            href="/dc"
          >
            DC
          </a>
          <a
            class="nav-item nav-link"
            href="/search"
          >
            Search
          </a>
        </div>
      </div>
      <div
        class="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end "
      >
        <ul
          class="navbar-nav ml-auto"
        >
          <span
            class="nav-item nav-link text-warning"
          >
            Batman
          </span>
          <button
            class="btn nav-item nav-link"
          >
            Logout
          </button>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
